.github/workflows/rust.yml

```diff
@@ -135,17 +132,26 @@ jobs:
       run: echo "${{ matrix.TARGET }}"
 
     - uses: actions/checkout@master
-    - uses: actions-rs/toolchain@v1.0.1
-      with:
-        toolchain: stable
-        target: ${{ matrix.TARGET }}
-        override: true
-
-    - uses: actions-rs/cargo@v1
-      with:
-        use-cross: true
-        command: build
-        args: --verbose --release --target=${{ matrix.TARGET }}
+    - name: Install build dependencies - Rustup
+      run: |
+        curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- --default-toolchain stable --profile default --target ${{ matrix.TARGET }} -y
+        echo "$HOME/.cargo/bin" >> $GITHUB_PATH
```

Comment:
Why this instead of using `actions-rs/toolchain`?

Reply:
This action is deprecated/archived and the usage is not recommended. 
![image](https://github.com/user-attachments/assets/f559793e-a00f-4907-bce4-4961b29a22f8)

Reply:
@patrickelectric did you consider using `dtolnay/rust-toolchain`? Its what I've done in my own little hotfix.

Reply:
Thanks @9SMTM6! Ill take a look.

Reply:
@patrickelectric  I'm currently testing a pratty big rewrite of this toolchain on my app. So perhaps wait.

One of the core issues was the heavy usage of cross. I was able to remove all usages of cross other than for arm64 linux. The other was that this PR still leaves A LOT of uses of actions-rs in the pipeline. A bunch of toolchain commands, which I had removed earier in my pipeline, but also a bunch of action-rs/cargo etc actions, I have overlooked these in my own toolchain too.